src/Control/Director.php

```diff
@@ -120,112 +117,40 @@ class Director implements TemplateGlobalProvider
      *
      * @uses handleRequest() rule-lookup logic is handled by this.
      * @uses TestController::handleRequest() This handles the page logic for a Director::direct() call.
-     * @param string $url
-     * @param DataModel $model
+     * @param HTTPRequest $request
+     * @return HTTPResponse
      * @throws HTTPResponse_Exception
      */
-    public static function direct($url, DataModel $model)
+    public static function direct(HTTPRequest $request)
     {
         // check allowed hosts
-        if (getenv('SS_ALLOWED_HOSTS') && !Director::is_cli()) {
-            $all_allowed_hosts = explode(',', getenv('SS_ALLOWED_HOSTS'));
-            if (!in_array(static::host(), $all_allowed_hosts)) {
-                throw new HTTPResponse_Exception('Invalid Host', 400);
-            }
-        }
-
-
-        // Validate $_FILES array before merging it with $_POST
-        foreach ($_FILES as $k => $v) {
-            if (is_array($v['tmp_name'])) {
-                $v = ArrayLib::array_values_recursive($v['tmp_name']);
-                foreach ($v as $tmpFile) {
-                    if ($tmpFile && !is_uploaded_file($tmpFile)) {
-                        user_error("File upload '$k' doesn't appear to be a valid upload", E_USER_ERROR);
-                    }
-                }
-            } else {
-                if ($v['tmp_name'] && !is_uploaded_file($v['tmp_name'])) {
-                    user_error("File upload '$k' doesn't appear to be a valid upload", E_USER_ERROR);
-                }
+        if (getenv('SS_ALLOWED_HOSTS') && !static::is_cli()) {
```

Comment:
Sweet, thanks :) We can make it more flexible later on, it's not going to be the only/last feature which is prone to be moved to a middleware